dashboard/modules/job/job_manager.py

```diff
@@ -269,7 +284,30 @@ def _get_current_node_resource_key(self) -> str:
                         return key
         else:
             raise ValueError(
-                "Cannot found the node dictionary for current node.")
+                "Cannot find the node dictionary for current node.")
+
+    def _handle_supervisor_started(self, job_id: str,
+                                   result: Optional[Exception]):
+        if result is None:
+            return
+        elif isinstance(result, RuntimeEnvSetupError):
+            logger.info(f"Failed to set up runtime_env for job {job_id}.")
+            self._status_client.put_status(
+                job_id,
+                JobStatusInfo(
+                    status=JobStatus.FAILED,
+                    message=(f"runtime_env setup failed: {result}")))
+        elif isinstance(result, Exception):
+            logger.error(
+                f"Failed to start supervisor for job {job_id}: {result}.")
+            self._status_client.put_status(
+                job_id,
+                JobStatusInfo(
+                    status=JobStatus.FAILED,
+                    message=f"Error occurred while starting the job: {result}")
+            )
+        else:
+            assert False
```

Comment:
add a message here indicating we should never reach this line ?

Reply:
yup